numba/experimental/jitclass/decorators.py

```diff
@@ -4,30 +4,83 @@
 from numba.experimental.jitclass.base import register_class_type, ClassBuilder
 
 
-def jitclass(spec):
+def jitclass(cls_or_spec=None, spec=None):
     """
-    A decorator for creating a jitclass.
+    A function for creating a jitclass.
+    Can be used as a decorator or function.
 
-    **arguments**:
+    Different use cases will cause different arguments to be set.
 
-    - spec:
-        Specifies the types of each field on this class.
-        Must be a dictionary or a sequence.
-        With a dictionary, use collections.OrderedDict for stable ordering.
-        With a sequence, it must contain 2-tuples of (fieldname, fieldtype).
+    If specified, ``spec`` gives the types of class fields.
+    It must be a dictionary or sequence.
+    With a dictionary, use collections.OrderedDict for stable ordering.
+    With a sequence, it must contain 2-tuples of (fieldname, fieldtype).
 
-    **returns**:
+    Any class annotations for field names not listed in spec will be added.
+    For class annotation `x: T` we will append `("x", numba.typeof(T))` to
```

Comment:
isn't it `as_numba_type(T)` not `typeof(T)`?

Reply:
Yup. Updated.